liquigraph-core/src/main/java/org/liquigraph/core/io/lock/LockableConnection.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.core.io.lock;
+
+import org.liquigraph.core.exception.LiquigraphLockException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.UUID;
+
+public final class LockableConnection implements ConnectionSupplier {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LockableConnection.class);
+    private final Connection connection;
+    private final UUID uuid;
+    private final Thread task;
+
+    public LockableConnection(Connection connection) {
+        this.connection = connection;
+        this.uuid = UUID.randomUUID();
+        this.task = new Thread(new ShutdownTask());
+        initiateLocking();
+    }
+
+    private final void initiateLocking() {
+        registerLockRemovalHook();
+        ensureLockUnicity();
+        tryWriteLock();
+    }
+
+   /**
+    * Closes the underlying connection
+    */
+   @Override
```

Comment:
Nitpicking: the indentation is inconsistent in the 4 lines above. Happy new year, though :)